src/extension/control-flow/src/main/java/org/geoserver/flow/config/DefaultControlFlowConfigurator.java

```diff
@@ -195,4 +202,53 @@ public long getTimeout() {
         return timeout;
     }
 
+    @Override
+    public List<Resource> getFileLocations() throws IOException {
+        List<Resource> configurationFiles = new ArrayList<>();
+        GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
+        if (loader != null) {
+            Resource controlflow = loader.get(PROPERTYFILENAME);
+        
+            configurationFiles.add(controlflow);
+        } else if (this.configFile != null && this.configFile.getResource() != null) {
+            configurationFiles.add(this.configFile.getResource());
+        }
+        return configurationFiles;
+    }
+
+    @Override
+    public void saveConfiguration(GeoServerResourceLoader resourceLoader) throws IOException {
+        GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
+        if (loader != null) {
+            for(Resource controlflow : getFileLocations()) {
+                Resource targetDir = 
+                        Files.asResource(resourceLoader.findOrCreateDirectory(Paths.convert(loader.getBaseDirectory(), controlflow.parent().dir())));
+                
+                Resources.copy(controlflow.file(), targetDir);
+            }
+        } else if (this.configFile != null && this.configFile.getResource() != null) {
+            Resources.copy(this.configFile.getFile(), Files.asResource(resourceLoader.getBaseDirectory()));
+        } else if (this.configFile != null && this.configFile.getProperties() != null) {
+            File controlFlowConfigurationFile = Resources.file(resourceLoader.get(PROPERTYFILENAME), true);
+            OutputStream out = Files.out(controlFlowConfigurationFile);
+            try {
+                this.configFile.getProperties().store(out, "");
+            } finally {
+                out.flush();
+                out.close();
+            }
+        }
+    }
+
+    @Override
+    public void loadConfiguration(GeoServerResourceLoader resourceLoader) throws IOException {
+        synchronized (this) {
+            Resource controlflow = resourceLoader.get(PROPERTYFILENAME);
+            if (Resources.exists(controlflow)) {
+                configFile = new PropertyFileWatcher(controlflow);
+                configFile.setKnownLastModified(System.currentTimeMillis());
```

Comment:
Ah, I see you're doing it because a test is using a fixedwatcher. The consideration about isStale is still valid though.